src/border_agent/border_agent.cpp

```diff
@@ -210,6 +215,63 @@ void BorderAgent::Stop(void)
     UnpublishMeshCopService();
 }
 
+#if OTBR_ENABLE_EPSKC
+void BorderAgent::HandleEpskcStateChanged(void *aContext)
+{
+    BorderAgent* borderAgent = static_cast<BorderAgent*>(aContext);
+    if (otBorderAgentIsEphemeralKeyActive(borderAgent->mNcp.GetInstance())) {
+        borderAgent->PublishEpskcService();
+    } else {
+        borderAgent->UnpublishEpskcService();
+    }
+}
+
+void BorderAgent::PublishEpskcService() {
+
+    otInstance              *instance = mNcp.GetInstance();
+    int                      port     = otBorderAgentGetUdpPort(instance);
+    Mdns::Publisher::TxtData txtData;
+
+    otbrLogInfo("Publish meshcop-e service %s.%s.local. port %d", mServiceInstanceName.c_str(), kBorderAgentEpskcServiceType, port);
+
+    mPublisher.PublishService(/* aHostName */ "", mServiceInstanceName, kBorderAgentEpskcServiceType,
+                              Mdns::Publisher::SubTypeList{}, port, txtData, [this](otbrError aError) {
+                                  if (aError == OTBR_ERROR_ABORTED)
+                                  {
+                                      // OTBR_ERROR_ABORTED is thrown when an ongoing service registration is
+                                      // cancelled. This can happen when the meshcop service is being updated
+                                      // frequently. To avoid false alarms, it should not be logged like a real error.
+                                      otbrLogInfo("Cancelled previous publishing meshcop-e service %s.%s.local",
+                                                  mServiceInstanceName.c_str(), kBorderAgentEpskcServiceType);
+                                  }
+                                  else
+                                  {
+                                      otbrLogResult(aError, "Result of publish meshcop-e service %s.%s.local",
+                                                    mServiceInstanceName.c_str(), kBorderAgentEpskcServiceType);
+                                  }
+                                  if (aError == OTBR_ERROR_DUPLICATED)
```

Comment:
Nit: this `if` is not logically relevant to the `if {} else {}` above it, so better adding a new line to split the two parts clearerly.
```suggestion
                                  
                                  if (aError == OTBR_ERROR_DUPLICATED)
```

Reply:
Thanks, accept.